packages/stream-model/src/model.ts

```diff
@@ -158,63 +68,146 @@ async function throwReadOnlyError(): Promise<void> {
   )
 }
 
+/**
+ * Represents the relationship between an instance of this model and the controller account.
+ * 'list' means there can be many instances of this model for a single account. 'link' means
+ * there can be only one instance of this model per account (if a new instance is created it
+ * overrides the old one).
+ */
+export enum ModelAccountRelation {
+  LIST = 'list',
+  LINK = 'link',
+}
+
+/**
+ * Contents of a Model Stream.
+ */
+export interface ModelDefinition {
+  name: string
+  description?: string
+  schema: JSONSchema.Object
+  accountRelation: ModelAccountRelation
+}
+
 /**
  * Model stream implementation
  */
 @StreamStatic<StreamConstructor<Model>>()
-export class Model<T = Record<string, any>> extends Stream {
+export class Model extends Stream {
   static STREAM_TYPE_NAME = 'model'
   static STREAM_TYPE_ID = 2
 
+  // The hardcoded "model" StreamID that all Model streams have in their metadata. This provides
+  // a "model" StreamID that can be indexed to query the set of all published Models.
+  private static readonly MODEL: StreamID = (function () {
+    const data = encode('model-v1')
+    const multihash = multihashes.encode(data, 'identity')
+    const digest = create(code, multihash)
+    const cid = CID.createV1(code, digest)
+    return new StreamID(Model.STREAM_TYPE_ID, cid)
```

Comment:
@oed not sure if I did this right or how you had been thinking, please let me know if there's a better/different way to construct the meta-model StreamID

Reply:
This is what I had in mind yes 👍